### DISCUSSION TITLE
Is it possible to verify the author when submitting code to GitHub

### DISCUSSION THREAD

**Discussion by chunlongniu** [community/community#123213] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I created a new project on GitHub on my personal laptop and would like to share some editor configurations with someone who uses the same editor. I am both the author and a user of this project. After pulling down the source code to my company's laptop to use it, unfortunately, some issues occurred while coding. I fixed them and pushed the changes to GitHub. However, when I reviewed my source code on GitHub, I noticed that the author is listed as the company's email address instead of my personal email address, even though I used my GitHub account to push the source code. I would like to know if there is any way to verify the author of a commit when pushing code to GitHub.

**Comment by KrlitosForever** [comment, marked as answer]
Yes, GitHub allows you to attribute authorship to specific individuals when submitting code through various mechanisms. When you commit changes to a GitHub repository, you're typically prompted to provide a username and email address, which are associated with your account. This information is recorded in the commit history, providing a record of who contributed what changes to the codebase.

Additionally, GitHub supports the concept of "signing" commits and tags using GPG (GNU Privacy Guard) keys. By signing your commits with your GPG key, you can cryptographically verify that the commit was made by you and hasn't been tampered with. This can add an extra layer of verification to the authorship of the code.

Overall, while GitHub provides mechanisms for attributing authorship to individuals and verifying the authenticity of commits, it's important to note that it ultimately relies on users to provide accurate information.

It's always better to create a branch so that you can verify it later, and if you agree, apply it to the main branch. It's not recommended to apply changes directly to the 'main' branch. Similarly, you can always go back in time; that's the magic of using Git and GitHub.

**Reply by chunlongniu** [reply]
got it, thank you very much

**Reply by ebndev** [reply]
@KrlitosForever thanks for taking the time to answer posts in Discussions 😄

**Reply by KrlitosForever** [reply]
Thank you very much, I try to help as much as I can :D
